### DISCUSSION TITLE
Importing data exported from Dashlane

### DISCUSSION THREAD

**Discussion by yannduran** [dani-garcia/vaultwarden#1969] [Q&A]
I'm running Vaultwarden in a linux docker container. And I had to install the old (deprecated) Dashlane desktop app to be able to export to JSON. The web extension now only exports to CSV.

Will there be an option to import CSV data in the future?

Anyway, somehow when trying to import data exported from Dashlane I've ended with duplicates of everything. I've checked the JSON and there doesn't seem to be any duplicates there. When I tried importing from a JSON **file** I got time-out errors twice. Then eventually I just pasted the JSON **text** in the box and did it that way.

If I click `Select All`, then `Delete selected` it asks me if I want to delete all 500 (although I thought there was around 650). The I click `Delete` and nothing seems to happen. I've tried it several times, and I've even left it sitting for a while. 

I tested selecting just a few and that worked fine. So I'm wondering if there's some kind of maximum number that can be bulk-deleted?

**Comment by BlackDex** [comment]
Import (and export) is fully done by the clients, nothing happens server side.
Since Vaultwarden only is the backend server it's not something we can fix on our side.

I see that there already was an csv importer, but it doesn't seem to be enabled or made it into the actual code, or has been removed.
https://github.com/bitwarden/web/blob/8e3d5b99c52db0ea47dbc1c0fdf5ef9337a9dfb8/src/app/tools/import.component.html

I suggest to either fill in a bug report here: https://github.com/bitwarden/web/issues
Or create a feature request here: https://community.bitwarden.com/c/feature-requests/5

The timeout has probably to do with you reverse proxy. I think the first import did worked, but took a long time for both Bitwarden web-vault and Vaultwarden to process which could result in the reverse proxy to generate a timeout, but Vaultwarden was still working on importing everything.

Regarding the delete, i have no clue, didn't come across it my self yet.

**Comment by yannduran** [comment]
>I see that there already was an csv importer, but it doesn't seem to be enabled or made it into the actual code

Hopefully it will get into the code.

>The timeout has probably to do with you reverse proxy

Yes, your explanation makes perfect sense, so thanks for that!

>Regarding the delete, i have no clue, didn't come across it my self yet.

Hmm, thanks anyway. As I'd only just started using Vaultwarden I ended up nuking the container's data and started again from scratch.
